descriptors/descriptors_test.go

```diff
@@ -1,125 +1,106 @@
+	vectors, err := ReadDerivationTestVectors()
+	require.NoError(t, err)
+
+	for _, vector := range vectors {
+		t.Run(vector.Name, func(tt *testing.T) {
+			descriptor, err := NewDescriptor(vector.Descriptor)
+
+			if vector.ExpectErr != "" {
+				require.ErrorContains(tt, err, vector.ExpectErr)
+				return
+			}
+
+			require.NoError(t, err)
+			defer descriptor.Close()
+
+			require.EqualValues(
+				t, vector.NumMultipath,
+				descriptor.MultipathLen(),
+			)
+
+			expectedDescriptor := vector.Descriptor
+			if !vector.HasChecksum {
+				expectedDescriptor += vector.Checksum
+			}
+			require.Equal(
+				t, expectedDescriptor, descriptor.String(),
+			)
+
+			for _, addr := range vector.Addresses {
+				network, err := ParseNetwork(addr.Network)
+				require.NoError(t, err)
+
+				parsedAddr, err := descriptor.AddressAt(
+					network, addr.MultipathIndex,
+					addr.DerivationIndex,
+				)
+
+				if addr.ExpectErr != "" {
+					require.ErrorContains(
+						tt, err, addr.ExpectErr,
+					)
+					continue
+				}
+
+				require.NoError(t, err)
+				require.Equal(t, addr.Address, parsedAddr)
+			}
+		})
+	}
 }
 
-func TestNetworksTr(t *testing.T) {
-	descriptorStr := "tr(xpub6BzikmgQmvoYG3ShFhXU1LFKaUeU832dHoYL6ka9JpCqKXr7PTHQHaoSMbGU36CZNcoryVPsFBjt9aYyCQHtYi6BQTo6VfRv9xVRuSNNteB/<0;1>/*)"
-	checksum := "#rq20sev9"
-	descriptor, err := NewDescriptor(descriptorStr)
-	require.NoError(t, err)
-	defer descriptor.Close()
-
-	require.Equal(t, descriptorStr+checksum, descriptor.String())
-
-	addr, err := descriptor.AddressAt(NetworkMainnet, 0, 0)
-	require.NoError(t, err)
-	require.Equal(t, "bc1puc00mgvj7hh25y8uhpux7gudjp9a986awgtemtxdfx6kcv4dkl5qt275jk", addr)
-
-	addr, err = descriptor.AddressAt(NetworkTestnet, 0, 0)
-	require.NoError(t, err)
-	require.Equal(t, "tb1puc00mgvj7hh25y8uhpux7gudjp9a986awgtemtxdfx6kcv4dkl5quzgmge", addr)
-
-	addr, err = descriptor.AddressAt(NetworkRegtest, 0, 0)
-	require.NoError(t, err)
-	require.Equal(t, "bcrt1puc00mgvj7hh25y8uhpux7gudjp9a986awgtemtxdfx6kcv4dkl5q3mzaar", addr)
+type TestAddress struct {
+	Network         string `json:"network"`
+	MultipathIndex  uint32 `json:"multipath_index"`
+	DerivationIndex uint32 `json:"derivation_index"`
+	Address         string `json:"address"`
+	ExpectErr       string `json:"expect_err"`
 }
 
-func TestInvalidChecksum(t *testing.T) {
-	_, err := NewDescriptor("tr([e81a5744/48'/0'/0'/2']xpub6Duv8Gj9gZeA3sUo5nUMPEv6FZ81GHn3feyaUej5KqcjPKsYLww4xBX4MmYZUPX5NqzaVJWYdYZwGLECtgQruG4FkZMh566RkfUT2pbzsEg/<0;1>/*,and_v(v:pk([3c157b79/48'/0'/0'/2']xpub6DdSN9RNZi3eDjhZWA8PJ5mSuWgfmPdBduXWzSP91Y3GxKWNwkjyc5mF9FcpTFymUh9C4Bar45b6rWv6Y5kSbi9yJDjuJUDzQSWUh3ijzXP/<0;1>/*),older(65535)))#lg9nqqha")
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "Invalid checksum")
+type DerivationTestVector struct {
+	Name         string         `json:"name"`
+	Descriptor   string         `json:"descriptor"`
+	HasChecksum  bool           `json:"has_checksum"`
```

Comment:
json files are normally camelCase, aren't they? So `hasChecksum`, `numMultipath`, `expectedErr`.

Reply:
I guess it comes down to personal preference and/or a project's accepted style. But changed it to camelCase (which also means we don't need the `json` struct tags).